plugins/hls-eval-plugin/src/Ide/Plugin/Eval/Parse/Option.hs

```diff
@@ -24,10 +21,13 @@ Right []
 >>> langOptions ""
 Left "No match"
 -}
-langOptions :: [Char] -> Either String [[Char]]
-langOptions = runParser (many space *> languageOpts <* many space)
+langOptions :: String -> Either String [String]
+langOptions =
+  left errorBundlePretty
+  . parse (space *> languageOpts <* eof) ""
 
--- >>> runParser languageOpts ":set -XBinaryLiterals -XOverloadedStrings"
--- Right ["BinaryLiterals","OverloadedStrings"]
-languageOpts :: Parser Char [[Char]]
-languageOpts = string ":set" *> many (many space *> string "-X" *> many letterChar)
+-- >>> parseMaybe languageOpts ":set -XBinaryLiterals -XOverloadedStrings"
+-- Just ["BinaryLiterals","OverloadedStrings"]
+languageOpts :: Parsec Void String [String]
+languageOpts = string ":set" *> space1
+  *> many (string "-X" *> many letterChar <* space)
```

Comment:
This does not support, for example, `Rank2Types`, right? Is this intentional?

Reply:
This code remains the same as old Extended Eval Plugin, except switched to use Megaparsec as a parsing backend. That is to say, this is so regardless of my intension from the start and already exists in the current master. Anyway, yes, I think we must use GHC's flag parser (if exists) to allow more flags of form e.g. `-fprint-explicit-foralls` etc.

Reply:
Aha, thank you for the explanation.